### ISSUE TITLE
Understand how the pastebin API works

### ISSUE THREAD

**Issue by xofbd** [xofbd/pastebin#3]
we want to get the post content and their meta data. We need to check out the [API documentation](https://pastebin.com/doc_api) to understand what endpoints we need to use. Once we've figured that out, we can start coding that up!

**Comment by i02132002** [comment]
To make requests to the API, you need to first create an account and login, then a unique developer API key will be provided to you. Since we probably don't want to include our own developer API keys in the repository, maybe we need a piece of software that lets the user input their developer API key and stores it locally. Is there a way to not store API keys in plaintext?

**Comment by i02132002** [comment]
Hey @shakibaRafiee, @sarvenaz-ch, @xofbd, unfortunately, I think I found a fatal flaw in the project. Pastebin only allows a user to look at their own pastes through their API. They enforce this by making you generate a session key with your login and password before you can request a paste. Your session key only gives you authorization to view your own pastes. I tested it out with some minimal code (I'd be happy to send it around). To scrape their data, they actually want you to use their scraping API, which is only available to "pro" users, and they aren't selling those accounts anymore, so there's no practical way to scrape their data. Also, if we try to scrape from the recent pastes page (https://pastebin.com/archive), there are only 50 of them, and if we scrape too often they will block our IPs. What do we do? ¯\_(ツ)_/¯

**Comment by xofbd** [comment]
I swear I didn't see this back in May when I first thought about this idea. what do others think?

We then scrape!

**Comment by xofbd** [comment]
@i02132002 Where did you info about blocking IP? I can't seem to find a `robot.txt`. Our scraper would probably only hit the site a few times a day, at least the `/archive` endpoint.

**Comment by i02132002** [comment]
Yes, I agree we'll probably only hit the site a few times a day to get the most recent 50 archived pastes, but we'll then need to scrape those 50. Not sure if that'll get us blocked?

They're saying on this page (https://pastebin.com/doc_scraping_api) that: "Important: Please make sure you ONLY fetch the scraping API endpoints listed on this page. If you scrape our website (including /raw/* pages) with your whitelisted IP, you will get blocked."

**Comment by xofbd** [comment]
maybe I'm taking this too literally:
>  with your whitelisted IP
What if we never got our IP whitelisted? Then we're cleared, right? :smiley:

**Comment by i02132002** [comment]
Yes, I think that will work, certainly worth trying. They used to sell the pro accounts for $20 with lifetime access. Now they seem to be considering moving to a subscription based model.

Apparently, pastebin is frequently used by malware to download malicious payload onto target computers, which makes the pastebin API very interesting and valuable to the security community!
https://www.reddit.com/r/netsec/comments/g20oxt/pastebin_scraping_api_removed/

**Comment by i02132002** [comment]
Ok, new plan of action then: instead of using their API, we need to scrape the /archive page, but we'll certainly be banned if they detect we're scraping too often. Looks like on average there are about 5 pastes per hour, and given 50 pastes total, if we scrape every 5 hours we'll be able to safely catch everything and not be banned. So here's my proposal:

1. Go to pastebin.com/archive and run a python script to scrape 50 paste keys every 5 hours.
2. Check which paste keys are new and check them at pastebin.com/mypastekey to scrape the contents of the pastes (probably no faster than one paste every 5 seconds).
3. If 2 fails, then we need to change our IP through a free VPN service (maybe we can use this: gist.github.com/Lazza/bbc15561b65c16db8ca8) and split up our workers in parallel so they come from different IPs.
